packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -85,9 +85,9 @@ export const AccountPageList: FunctionComponent<AccountPageListProps> = ({
   const [offsetLeft, setOffsetLeft] = useState(0);
 
   const [treeItems, setTreeItems] = useState<TreeItem[]>([]);
-  const [prevData, setPrevData] = useState(data);
+  const [prevData, setPrevData] = useState<typeof data>([]);
 
-  if (data !== prevData && !loading) {
+  if (!loading && data !== prevData) {
     setPrevData(data);
     setTreeItems(getTreeItemList(data));
   }
```

Comment:
Would a better solution look something like this? Makes the intention clearer to me? 

```ts
const [treeItems, setTreeItems] = useState(() => getTreeItemList(data));
const [prevData, setPrevData] = useState(data);

if (data !== prevData) {
  setPrevData(data);
  setTreeItems(getTreeItemList(data));
}
```

Reply:
Followed your suggestion in eaec4f4a103763969d27da41c15684e54b238b2c.